DataFormats/TrackerCommon/interface/LegacyTrackerTopology.h

```diff
@@ -0,0 +1,98 @@
+#ifndef TRACKER_TOPOLOGY_LEGACY_H
+#define TRACKER_TOPOLOGY_LEGACY_H
+
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/DetId/interface/DetId.h"
+
+#include <memory>
+
+// WARNING: this header has been introduced to call a TrackerTopology
+// object whenever it is not possible to access it through an EventSetup.
```

Comment:
what are the cases where the EventSetup can not deliver this?
TrackerTopologyEP and relevant upstream can be set up to provide the phase-0 topology just as it is done in normal running.